src/DotVVM.Framework/ViewModel/Serialization/ViewModelTypeMetadataSerializer.cs

```diff
@@ -163,6 +163,10 @@ internal JToken GetTypeIdentifier(Type type, HashSet<Type> dependentObjectTypes,
             {
                 return new JObject(new JProperty("type", "dynamic"));
             }
+            else if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Dictionary<,>))
```

Comment:
I'd rather check if the type implements `IDictionary<,>` - not sure if all people use concrete `Dictionary`.